plugins/org.locationtech.udig.render.gridcoverage.basic/src/org/locationtech/udig/render/internal/gridcoverage/basic/GridCoverageReaderRenderer.java

```diff
@@ -111,6 +111,11 @@ public GridCoverageReaderRenderer(){
     public synchronized void render( Graphics2D graphics, IProgressMonitor monitor )
             throws RenderException {
         try {
+                // faster rendering if out of scale
+                GridCoverageRenderState state = getRenderState(getContext());
+                double scale = state.context.getViewportModel().getScaleDenominator();
+                if (scale < state.minScale || scale > state.maxScale)
```

Comment:
Great, change looks good! Any chance to add a test case?

Reply:
IMH minScale / maxScale of state object might be null, if one or the other is not set.

Can you explaint how this change is related to line 232/ 227 where Rule of Style is compared to teh currentScale? I guess it is possible to define different rules for the same resource and one rule might fit where another doesn't.

So maybe it makes sense to move the style rule scale comparence before creating a coverage ..